src/DIPS.Xamarin.UI/Controls/Popup/PopupBehavior.cs

```diff
@@ -134,29 +136,88 @@ private void HidePopup()
                 return;
             }
 
+
             var layout = m_attachedTo.GetParentOfType<PopupLayout>();
             if (layout == null) throw new InvalidProgramException("Can't have a popup behavior without a PopupLayout around the element");
             layout.HidePopup();
         }
     }
 
+    /// <summary>
+    /// Horizontal location of the popup relative to the parent item
+    /// </summary>
+    public enum HorizontalPopupOptions
+    {
+        /// <summary>
+        /// Left side of the popup is aligned with the left side of the parent.
+        /// </summary>
+        LeftAlign,
+        /// <summary>
+        /// Right side of the popup is aligned with the right side of the parent.
+        /// </summary>
+        RightAlign,
+        /// <summary>
+        /// Popup is centered above the parent.
+        /// </summary>
+        Center,
+        /// <summary>
+        /// Popup is placed all the way to the left of the parent.
+        /// </summary>
+        Left,
+        /// <summary>
+        /// Popup is placed all the way to the right of the parent.
+        /// </summary>
+        Right
+    }
+
+    /// <summary>
+    /// Vertical orientation of the popup relative to the parent item
+    /// </summary>
+    public enum VerticalPopupOptions
+    {
+        /// <summary>
+        /// Automatically selects above or below based on parent placement.
+        /// </summary>
+        Auto,
+        /// <summary>
+        /// Popup is placed above the parent.
+        /// </summary>
+        Above,
+        /// <summary>
+        /// Popup is placed below the parent.
+        /// </summary>
+        Below,
+        /// <summary>
+        /// Popup is placed on top of the parent.
+        /// </summary>
+        Center,
+    }
+
     /// <summary>
     /// Directions of the popup
     /// </summary>
+    [Obsolete("Use VerticalPopupOptions and HorizontalPopupOptions instead.")]
     public enum PopupDirection
     {
         /// <summary>
-        /// Automatically based on the location at the screen
+        /// None is the default and enables the usage of the Vertidal/Horizontal-Options.
```

Comment:
What does this mean? If I set it to none it will enable vertical horizontal? Not sure if I understand what it means